### DISCUSSION TITLE
I can't use HttpOnly:true option setting cookie

### DISCUSSION THREAD

**Discussion by bellasimi** [vercel/vercel#8732] [Help] [answered]
I deploy web projecty on Vercel. everything works well except set-cookie. 
this is my code. 

```js
import { Cookies } from 'react-cookie'

const cookies = new Cookies()

cookies.set(TOKEN_KEY, token, {
  path: '/',
  httpOnly: true,
  expires: new Date(expirationDate)
})
```

when i remove httpOnly option it works, but with that option it doesn't works. httpOnly option doesn't works in local server but Vercel app server is not local right? 

can you tell me how to fix it?

**Comment by amyegan** [comment]
Sorry you're running into this problem @bellasimi. Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can take a closer look at what is going wrong?

**Comment by bellasimi** [comment]
you mean the example code help test? 😮 
then **this is my git and you can see that problem in `refactor/cookie-httponly`**
**repo**
  https://github.com/prgrms-web-devcourse/Team-Tasty-Masiottae-FE.git
**pr**
  https://github.com/prgrms-web-devcourse/Team-Tasty-Masiottae-FE/pulls
**deploy  preview**
   https://team-tasty-masiottae-fe-git-refactor-cookie-httponly-tastyfe.vercel.app/

**Comment by amyegan** [comment, marked as answer]
Thanks for sharing the code and preview deployment! Can you share more details about what is not working when you attempt to set an httpOnly cookie? It will help me to know which steps to follow and what error I should expect to see.

**Reply by bellasimi** [reply]
ok. ☺above code( including logic setting cookie with httponly: true option)was complied, run well but doesn't work actually.
When someone login,  app use react-query useLoginMutaion. ( u can see code below )
if response is successful, then app call setTokenData ( made with cookies.set code below ) and set cookie.
in this process, cookie isnt't set in application cookie.
**there's no error message. even after runngin setTokenData and reload logic, but no cookie**

**I think that's caused by httponly option. Because without that option cookie set well with same code** 

 **src > hooks > mutations > useLoginMutation.ts**

```js
const postLogin = async ({ email, password }: Params) => {
  const { data } = await axios.post<Data>('/login', { email, password })

  return data
}

export const useLoginMutation = () => {
  const router = useRouter()
  const [user, setUser] = useRecoilState<User>(currentUser)
  return useMutation(postLogin, {
    onSuccess: (data) => {
      const { image } = data.account
      setUser({
        ...user,
        ...data.account,
        image: image ?? DEFAULT_USER_IMAGE
      })
      setTokenData(data)
      router.reload()
    }
  })
}
```

**src >utils > cookie.ts**

```js
export const setToken = ({ token, expirationDate }: Token) => {
  setCookie(TOKEN_KEY, token, {
    path: '/',
    expires: new Date(expirationDate),
    httpOnly: true
  })
  setCookie(TOKEN_EXPIRE_DATE, expirationDate, {
    path: '/',
    expires: new Date(expirationDate),
    httpOnly: true
  })
}
```

thanks for helping me Amy 😇

**Reply by bellasimi** [reply]
> Thanks for sharing the code and preview deployment! Can you share more details about what is not working when you attempt to set an httpOnly cookie? It will help me to know which steps to follow and what error I should expect to see.

Amy I find my mistake... that happened because I made the logic which approach cookie with script. thanks for helping me.

**Reply by amyegan** [reply]
I'm very happy you found the solution!
